bin/maglev-irb

```diff
@@ -1,39 +1,22 @@
-#!/usr/bin/env maglev-ruby
-# -*- ruby -*-
+#!/usr/bin/env bash
 #
-# Note: This script will not work if maglev-ruby is not in your path.
-#     So you should add something like the following line to your .bashrc
-#     export PATH=$MAGLEV_HOME/bin:$PATH
-#
-#   irb.rb - intaractive ruby
-#     $Release Version: 0.9.5 $
-#     $Revision: 11708 $
-#     $Date: 2007-02-13 08:01:19 +0900 (Tue, 13 Feb 2007) $
-#     by Keiju ISHITSUKA(keiju@ruby-lang.org)
-#
-$-I.unshift File.dirname($0)
-
-def STDIN.tty?
-  true
-end
-
-# MAGLEV_DEBUG_IRB = true # uncomment here and in irb.rb to debug
-
-# def to_s
-#     "main"
-#     end
-
-require "irb"
-
-
-if __FILE__ == $0
-  IRB.start(__FILE__)
-else
-  # check -e option
-  if /^-e$/ =~ $0
-    IRB.start(__FILE__)
+# Find MAGLEV_HOME
+if [ -z "$MAGLEV_HOME" ]; then
+  if [ -x $PWD/gemstone/bin/startstone ]; then
+    # echo "using $PWD as MAGLEV_HOME"
+    export MAGLEV_HOME=$PWD
   else
-    IRB.setup(__FILE__)
-  end
-end
+    echo "$COMMAND: [Warning] \$MAGLEV_HOME is not set."
+    tree=$(cd `dirname "$0"/` ; cd .. ; pwd)
+    if [ -d "${tree}/gemstone" ]; then
+      export MAGLEV_HOME="${tree}"
+      echo "Assuming ${MAGLEV_HOME}"
+    else
+      echo "$COMMAND: [Error] Neither $PWD nor $tree are a valid MAGLEV_HOME directory"
+      echo "To fix this, set a valid MAGLEV_HOME in your .bashrc"
+      exit 1
+    fi
+  fi
+fi
 
+$MAGLEV_HOME/bin/maglev-ruby "$@" $MAGLEV_HOME/bin/maglev_irb.rb
```

Comment:
Two words "I forgot" ;^)    I'll update tomorrow when I'm back at gemtalk,  and helloooo to germany